api/turing/validation/validator.go

```diff
@@ -196,6 +196,81 @@ func checkDanglingRoutes(
 	}
 }
 
+func validateConditionOrthogonality(
+	sl validator.StructLevel,
+	fieldName string,
+	allRules models.TrafficRules,
+) {
+	trafficConditionsFieldMap := map[string][]string{}
+	trafficConditionsValueMap := map[string]map[string]*set.Set{}
+	for _, rule := range allRules {
+		trafficConditionsValueMap[rule.Name] = map[string]*set.Set{}
+		trafficConditionsFieldMap[rule.Name] = []string{}
```

Comment:
Do we need this data structure? Can we just rely on `trafficConditionsValueMap` instead?

```golang
for idx, rule1 := range allRules {
	rule1Fields, ok := trafficConditionsValueMap[rule1.Name] {
		for field, currSet := range rule1Fields {
...
```

This will also make the code more concise.